web/handler.go

```diff
@@ -97,7 +97,8 @@ func (u *webHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set(k, v)
 	}
 
-	if len(c.Users) == 0 {
+	// If OPTIONS method or none users configured then skip auth check
+	if r.Method == http.MethodOptions || len(c.Users) == 0 {
```

Comment:
One possible safe way to do this would be to check both the method and the URI.

```suggestion
	if (r.RequestURI == "/" && r.Method == http.MethodOptions) || len(c.Users) == 0 {
```

This way we only allow bypass for the typical browser [Preflight request](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing#Preflight_example).

Reply:
the additional is not needed. Or maybe I miss something?